cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/Storage.java

```diff
@@ -121,7 +122,7 @@ public static Storage fromJson(JsonObject json) {
                             .collect(Collectors.toMap(Map.Entry::getKey, e -> String.valueOf(e.getValue())));
 
             // match-expressions doesn't supported yet
-            List<LabelSelectorRequirement> matchExpressions = null;
+            List<LabelSelectorRequirement> matchExpressions = Collections.emptyList();
```

Comment:
We should check how the fabric8 behaves with an empty `matchExpressions`. Using `null`, it doesn't set this field in the K8S resource. @tombentley have you tested it ?

Reply:
No, must admit I have not tested it. I'd be surprised if it were a problem though. I guess as an alternative we could just use a `LabelSelectorBuilder`.

Reply:
Yes but in this case we can remove the above statement and just live with something like this : 

`storage.withSelector(new LabelSelector(null, matchLabels));`

in the following statement. I hope that findbugs won't complain in this way :-)